packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -377,6 +377,15 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                             let method: string;
                             if (this.hostPolicy.concurrentSnapshotFetch && !this.hostPolicy.summarizerClient) {
                                 const snapshotP = this.fetchSnapshot(snapshotOptions, tokenFetchOptions);
+                                snapshotP.catch(async (error) => {
+                                    const errorType = error.errorType;
+                                    // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have permissions
+                                    // permissions for the file or it was deleted. So the user will again try to fetch from cache on any failure in future.
+                                    if (errorType === DriverErrorType.authorizationError || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError) {
+                                        await this.cache.persistedCache.removeAllEntriesForDocId(this.documentId);
```

Comment:
This code is defiantly in the wrong place.
getWithRetryForTokenRefresh() above handles expired tokens that results in retry on 401/403. Only if retry failed, then we can safely say that user has no permissions and on future runs we should not use cache.

Reply:
Made changes but in case 404 we need to clear here only because we will not make a refresh token call.

Reply:
Why not move all this code below getWithRetryForTokenRefresh, i.e. wrap result of getWithRetryForTokenRefresh call and add .catch to its result, where we handle both cases?